src/main/java/net/prominic/groovyls/providers/CompletionProvider.java

```diff
@@ -292,86 +264,60 @@ private void populateItemsFromClassNode(ClassNode classNode, Position position,
-            CompletionItem item = CompletionItemFactory.createCompletion(field, field.getName(), astContext);
-
-            if (!field.isDynamicTyped()) {
-                item.setDetail(field.getType().getNameWithoutPackage());
+        });
+        items.addAll(fields, f -> {
+            String name = f.getName();
+            if (!name.startsWith(memberNamePrefix) || existingNames.contains(name)) return null;
+            existingNames.add(name);
+            CompletionItem item = CompletionItemFactory.createCompletion(f, f.getName(), astContext);
+            if (!f.isDynamicTyped()) {
+                item.setDetail(f.getType().getNameWithoutPackage());
             }
             return item;
-        }).collect(Collectors.toList());
-        items.addAll(fieldItems);
+        });
     }
 
-    private void populateItemsFromMethods(List<MethodNode> methods, String memberNamePrefix, Set<String> existingNames,
-                                          Completions items) {
-        List<CompletionItem> methodItems = methods.stream()
-                .filter(method -> {
-                    String methodName = method.getName();
-                    // overloads can cause duplicates
-                    if (methodName.startsWith(memberNamePrefix) && !existingNames.contains(methodName)) {
-                        existingNames.add(methodName);
-                        return !method.getDeclaringClass().isResolved() ||
-                               (method.getModifiers() & GroovyASTUtils.EXPANSION_MARKER) != 0 ||
-                               GroovyReflectionUtils.resolveMethodFromMethodNode(method, astContext).isPresent();
-                    }
-                    return false;
-                }).map(method -> {
-                    CompletionItem item = CompletionItemFactory.createCompletion(method, method.getName(), astContext);
-
-                    var details = getMethodNodeDetails(method);
-                    item.setLabelDetails(details);
+    private void populateItemsFromMethods(List<MethodNode> methods, String memberNamePrefix, Set<String> existingNames, Completions items) {
+        items.addAll(methods, method -> {
+            String name = method.getName();
+            if (!name.startsWith(memberNamePrefix) || existingNames.contains(name)) return null;
+            existingNames.add(name);
+            if ((method.getDeclaringClass().isResolved() &&
+                    (method.getModifiers() & GroovyASTUtils.EXPANSION_MARKER) == 0) ||
+                    GroovyReflectionUtils.resolveMethodFromMethodNode(method, astContext) == null) {
+                return null;
+            }
 
-                    return item;
-                }).collect(Collectors.toList());
-        items.addAll(methodItems);
+            CompletionItem item = CompletionItemFactory.createCompletion(method, method.getName(), astContext);
+            item.setLabelDetails(getMethodNodeDetails(method));
+            return item;
```

Comment:
can the sorting location be increased to the top of the list if the method (or class/field) is annotated with one of the groovy documentation annotations? if this was done it would mean stuff like `recipeBuilder()`, `add(...)`, and `removeAll()` would be at the top, instead of the current order.

![image](https://github.com/user-attachments/assets/932bff07-5a60-4a70-a5d8-aa8b6db0eb5f)